### ISSUE TITLE
Deprecation of functionality i.e createClass

### ISSUE THREAD

**Issue by Skizu** [adazzle/react-data-grid#951]
- Have you already searched for similar issues? Please help us out and double-check first!

- Also, please don't be that person who deletes this template. It's here for a reason.

- Thanks!

---

### Which version of React JS are you using?

✅ Officially supported ✅
- [ ] v15.4.x

⚠️ Not officially supported, expect warnings ⚠️
- [ ] v15.5.x
- [x] v15.6.x

☣️ Not officially supported, expect warnings and errors ☣️
- [ ] v16.x.x

---

### Which browser are you using?

✅ Officially supported ✅
- [ ] IE 9 / IE 10 / IE 11
- [ ] Edge
- [ ] Chrome

⚠️ Not officially supported, but "should work" ⚠️
- [ ] Firefox
- [ ] Safari

---

### I'm submitting a ...

- [ ] 🐛 Bug Report
- [x] 💡 Feature Request

> 👋 Need general support? Not sure about how to use React itself, or how to get started with the Grid?
> Please do not submit support request here. Instead see

> https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md

---

### Issue Details

```
Please include:
- What the current behavior is
- What the desired behvaior is
- (If Bug) Steps to reproduce the issue
- (If Feature) The motivation / use case for the feature

We especially love screenshots / videos of problems, and remember
The Best Issue Is A Pull Request™
```
Will this project be upgraded to the latest stable of react, or will this project be bound to v15.4.x?

**Comment by zuyetawarmatik** [comment]
I think it's time for the maintainer to create 3.0. With 2.0, the lib may need to import 'create-react-class' as dependency. Reference https://facebook.github.io/react/blog/2017/09/26/react-v16.0.html

**Comment by jpdriver** [comment]
hey @Skizu @zuyetawarmatik -- we're currently considering both an interim 2.0 release that fixes these issues, and a 3.0 version to be release in future.

**Comment by 29er** [comment]
Any progress on  a 2.0 release that fixes this ?

**Comment by diogofcunha** [comment]
Fixed in 3.0.8
